src/Pyz/Yves/ProductImageWidget/Theme/default/components/molecules/image-gallery/image-gallery.ts

```diff
@@ -7,14 +7,20 @@ export default class ImageGallery extends Component {
     protected thumbnailSlider: $;
     protected defaultImageUrl: string;
     protected currentSlideImage: HTMLImageElement;
+    protected thumbnailButtons: HTMLButtonElement[];
 
     protected readyCallback(): void {}
 
     protected init(): void {
         this.galleryItems = <HTMLElement[]>Array.from(this.getElementsByClassName(`${this.jsName}__item`));
         this.thumbnailSlider = $(`.${this.jsName}__thumbnails`);
-
         this.initializationSlider();
+
+        this.thumbnailButtons = Array.from(this.querySelectorAll(`.slick-slide`));
+        this.thumbnailButtons.map((button: HTMLButtonElement) => {
```

Comment:
Why map?

Reply:
Lets move it to map Events.
Do we need `thumbnailButtons` as separate propeerty?